quetz/tasks/mirror.py

```diff
@@ -357,24 +361,28 @@ def handle_batch(update_batch):
             return False
 
         for package_name, metadata in packages.items():
-            path = os.path.join(arch, package_name)
+            if (include and check_package_membership(package_name, package_list)) or (
```

Comment:
this if condition is really confusing. Is there a way to simplify?

Reply:
I actually put it inside a KMAP solver / boolean algebra calculator. It cannot be simplified further. It's basically the condition `(A and B) or (!A and !B)` where A is `include` and B is `check_package_membership`. This intelligently takes care of both `inclusion` and `exclusion` because depending on `include` as True/False,
we check the first term `(A and B) => B because A is already true`
OR the second term `(!A and !B) => !B because A was false and thus !A is true`.

At the end, we check either `B` (include) or `!B` (exclude) based on the state of `A` (which tells us what to do).
So for each package, we either enter the loop or do not enter it (based on `B` or `!B`)

Reply:
One way to simplify difficult if conditions is by splitting and using `continue` or `break`. Maybe that works here?